### ISSUE TITLE
add CREATE2/CREATE destination address to vmlogs

### ISSUE THREAD

**Issue by winsvega** [ethereum/go-ethereum#28375]
# Rationale

would be nice to have result address of create/create2 (especially) be printed out in vmlog
many times we try to create smth and then rever/oog 
but we need to know the code address to check that it is not present in the state
so currently we do it manually. 

# Implementation

Calculate the hash of create instruction code and print the address in vmlog
even if it end up not being deployed to the post state or removed from it 
it would be still possible to see at what address the code was expected to be created

**Comment by s1na** [comment]
Just to clarify, you mean something like this?

```js
{
      depth: 1,
      gas: 29596589,
      gasCost: 0,
      op: "CREATE",
      createAddr: "0xabba",
      pc: 883,
      refund: 2800,
      stack: ["0x2e353ada", "0x5b", "0x116", "0x2099b72bd243053674599a1a76053641352e2394", "0x1", "0x1", "0x0", "0x0"]
  }]
  ```

We could add this. However since the tracer hook is prior to opcode execution, we don't have a way to tell if the CREATE failed and would emit the address anyway.

**Comment by winsvega** [comment]
yes, if this does not break anything. all we need is just to see in the logs projected CREATE address. 
if it is not deployed does not matter. the address is calculated from init code and when this code is dynamic it is pain to calculate it.
